.circleci/config.yml

```diff
@@ -141,6 +141,26 @@ jobs:
           name: Upload Coverage
           command: bash <(curl -s https://codecov.io/bash) -cF python
 
+  build_image:
+    docker: 
+      - image: docker
+    steps:
+      - setup_remote_docker
+      - checkout
+      - run:
+          name: Docker Build
+          command: docker build -t prefecthq/prefect .
+      - run:
+          name: Test image
+          command: |
+            docker run -dit prefecthq/prefect /bin/bash -c 'curl -fL0 https://raw.githubusercontent.com/PrefectHQ/prefect/master/examples/retries_with_mapping.py | python'
+      - run:
+          name: Authenticate with Docker Hub and push
+          command: |
+            docker login --username  prefectci --password $DOCKER_HUB_PW
```

Comment:
Could a random person submit a PR with a new circle config that does
```
echo $DOCKER_HUB_PW
```
and see this password?

Reply:
CircleCI doesn't share environment variables to *forks*, so PRs from forks should be unable to do this. However, I suggest that we test this to be absolutely sure it works as expected.

Reply:
I was under the impression at the time that secrets do not interpolate for pull requests.  However, I've just looked it up again and Jeremiah is right--that's only FORKED PRs

Reply:
Ok cool, that should be sufficient.  After this is merged I'll fork and try to echo the username and make sure it doesn't work.

Reply:
since non-prefect folks cannot make non-forked PRs, I am going to assume this is safe (assuming it passes Chris's test to verify it works how we understand it to work)